pkg/registry/apis/provisioning/jobs.go

```diff
@@ -0,0 +1,110 @@
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	provisioning "github.com/grafana/grafana/pkg/apis/provisioning/v0alpha1"
+	"github.com/grafana/grafana/pkg/registry/apis/provisioning/jobs"
+)
+
+type jobsConnector struct {
+	repoGetter RepoGetter
+	jobs       jobs.Queue
+	historic   jobs.History
+}
+
+func (*jobsConnector) New() runtime.Object {
+	return &provisioning.Repository{}
+}
+
+func (*jobsConnector) Destroy() {}
+
+func (*jobsConnector) ProducesMIMETypes(verb string) []string {
+	return []string{"application/json"}
+}
+
+func (c *jobsConnector) ProducesObject(verb string) any {
+	return &provisioning.Job{}
+}
+
+func (*jobsConnector) ConnectMethods() []string {
+	return []string{http.MethodPost, http.MethodGet}
+}
+
+func (*jobsConnector) NewConnectOptions() (runtime.Object, bool, string) {
+	return nil, true, "" // path -> uid
+}
+
+func (c *jobsConnector) Connect(
+	ctx context.Context,
+	name string,
+	opts runtime.Object,
+	responder rest.Responder,
+) (http.Handler, error) {
+	repo, err := c.repoGetter.GetHealthyRepository(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	cfg := repo.Config()
+
+	return withTimeout(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
shouldn't we take the context from the request here so that when the requests gets cancelled that gets propagated?